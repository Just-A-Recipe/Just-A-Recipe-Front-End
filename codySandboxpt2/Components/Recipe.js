import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ image, title, ingredients, instructions, measures }) => (
  <figure>
    <img src={image}/>
    <figcaption>
      <h2>{title}</h2>
    </figcaption>
    <p>{ingredients}</p>
    <p>{measures}</p>
    <p>{instructions}</p>
  </figure>
);

//***Informaiton below is based off recipes/${id}/information?{&apiKey=blahblahblah}
//extendedIngredients: name & measures.us.amount & measures.us.unitLong  
//image: image
//title: title
//ingredients: extendedIngredients.name (multiple)
//measures: extendedIngredients.measures.us.amount & extendedIngredients.measures.us.unitLong
//instructions: instructions (will have to see how the data comes out as it have \n and others have br)
Recipe.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.object).isRequired, 
  measures: PropTypes.object.isRequired,
  instructions: PropTypes.string.isRequired
};

export default Recipe;
 
