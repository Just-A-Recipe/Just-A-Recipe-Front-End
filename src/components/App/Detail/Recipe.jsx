import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ image, title, ingredients, instructions, measures }) => (
  <figure>
    <img src={image}/>
    <figcaption>
      <h2>{title}</h2>
    </figcaption>
    <p>{ingredients}</p>
    <p>{measures}</p>
    <p>{instructions}</p>
  </figure>
);

Recipe.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  })),
  measures: PropTypes.object(PropTypes.shape({
    amount:PropTypes.string.isRequired,
    us: PropTypes.string.isRequired
  })),
  instructions: PropTypes.string.isRequired
};

export default Recipe;
//ingredients is an array of objs and also instructions. We have to figure out how to make thos proptypes. 
