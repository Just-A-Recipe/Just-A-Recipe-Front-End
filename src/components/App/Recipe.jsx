import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ image, title, ingredients, instructions, measures }) => (
  <figure>
    <img src={image}/>
    <figcaption>
      <h2>{title}</h2>
    </figcaption>
    <p>{ingredients}</p>
    <p>{measures}</p>
    <p>{instructions}</p>
  </figure>
);

Recipe.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.object).isRequired,
  measures: PropTypes.object.isRequired,
  instructions: PropTypes.string.isRequired
};

export default Recipe;
 
