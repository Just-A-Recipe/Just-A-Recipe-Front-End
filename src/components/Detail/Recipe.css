.mainTitle {
    font-weight: bold; 
    font-family: cursive;   
    font-size: 4.5em;
    margin-top: 17px;
    margin-bottom: 17px;
    text-align: center;
}
.titles {
    text-decoration: underline salmon;
    font-size: x-large;
}
.icon {
    width: 20px;
    position: relative;
}
.ingredients, .instructions {
    text-align: center;
    width: 35vw;
    font-size: large;
    padding-bottom: 5rem;
}
.imageDiv {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 50vh;
    margin-right: auto;
    margin-left: auto;
}
.image {
    border-radius: 20px;
    border-top: 20px solid #FFA384;
    border-right: 20px solid #EFE7BC;
    border-bottom: 20px solid #74BDCB;
    border-left: 20px solid #E7F2F8
}
.theSection {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin-top: 17px;
}
