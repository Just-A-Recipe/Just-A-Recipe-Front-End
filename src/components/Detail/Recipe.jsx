import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import imgIcon from '../../assets/favorite.png';
import secondFavIcon from '../../assets/star.png';
import {  withRouter, useParams } from 'react-router-dom';
import { addFavorite, deleteFavorite, getUserFavorites } from '../../services/spoonacular';
import firebase from '../Firebase/Firebase';
  
import styles from './Recipe.css';

const Recipe = ({ image, title, ingredients, instructions, id }) => {
  // console.log(image, title);
  const [message, setMessage] = useState('');
  const [favorite, setFavorite] = useState(false);
 
  useEffect(() => {
    getUserFavorites(firebase.getCurrentEmail())
      .then(fetchedFavorite => {
        console.log(fetchedFavorite);
        const isFavorite = fetchedFavorite.body.some(({ recipeId }) => recipeId === +id);
        setFavorite(isFavorite);
      });
  }, []);

  const instructionElements = instructions[0].steps.map(instruction => (
    <div key={instruction.id}>
      <p> {instruction.step} </p>
    </div>
  ));

  const ingredientsElements = ingredients.map(ingredient => (
    <div key={ingredient.id}>
      <p> {ingredient.original}</p>
    </div>
  ));
  const handleAddFavorite = (id) => {
    setFavorite(true);
    addFavorite(firebase.getCurrentEmail(), parseInt(id))
      .then(res => {
        if(!firebase.getCurrentEmail()){
          setMessage('Login / Create Account to Add to Favorites!');
        } else {
          setMessage(`${res.id} Added to Favorites!`);
        }
      }).catch(err => {
        console.error(err);
      });
  };
  const handleDeleteFavorite = (id) => {
    let favId;
    setFavorite(false);
    getUserFavorites(firebase.getCurrentEmail())
      .then((res)=> {
        res.body.forEach(fav => {
          if(fav.recipeId == id) {
            favId = fav._id;
          }
        });
        deleteFavorite(favId)
          .then(res => {
            setMessage(res.message);
          }).catch(err => {
            console.error(err);
          });
      });
  };

  // useEffect(() => {
  //   getUserFavorites(firebase.getCurrentEmail())
  //   setFavorite()
  // });
  
  return (
    <div>
      <div className={styles.imageDiv}>

        {!favorite && <img onClick={() => handleAddFavorite(id)} className={styles.icon} src={imgIcon} alt=''/>}
        {favorite && <img onClick={() => handleDeleteFavorite(id)} className={styles.icon} src={secondFavIcon} alt=''/>}

        {message}
        
        <img className={styles.image} src={`${image}`} />
        <h2 className={styles.mainTitle}>{title}</h2>
      </div>

      <section className={styles.theSection}>

        <div className={styles.ingredients}>
          <p className={styles.titles}>Ingredients</p>
          {ingredientsElements}
        </div>

        <div className={styles.instructions}>
          <p className={styles.titles}>Instructions</p>
          {instructionElements}
        </div>

      </section>
    </div>
  );
};

Recipe.propTypes = {
  id: PropTypes.string.isRequired,
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.object),
  instructions: PropTypes.arrayOf(PropTypes.object),
};

export default withRouter(Recipe);
