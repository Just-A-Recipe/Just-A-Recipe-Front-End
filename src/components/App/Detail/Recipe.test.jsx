import React from 'react';
import { shallow } from 'enzyme';
import Recipe from './Recipe';

describe('Recipe component', () => {
  it('renders a Recipe', () => {
    const wrapper = shallow(<Recipe
      title= 'spaghetti'
      image='https://www.inspiredtaste.net/wp-content/uploads/2019/03/Spaghetti-with-Meat-Sauce-Recipe-1-1200.jpg'
    />);
  });
});

//its not done we have to fixed the component 