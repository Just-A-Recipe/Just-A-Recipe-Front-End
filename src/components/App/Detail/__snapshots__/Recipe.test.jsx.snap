// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component renders a Recipe 1`] = `
<figure>
  <img
    src="https://www.inspiredtaste.net/wp-content/uploads/2019/03/Spaghetti-with-Meat-Sauce-Recipe-1-1200.jpg"
  />
  <figcaption>
    <h2>
      spaghetti
    </h2>
  </figcaption>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
  <p>
    some long instrcutions
  </p>
</figure>
`;
